### ISSUE TITLE
Convert IPv6 CIDR to IP / Netmask notation

### ISSUE THREAD

**Issue by pquerner** [mlocati/ip-lib#28]
Is it possible to get the netmask of a given ipv4 address with cidr notation?

**Comment by mlocati** [comment]
Do you have an example of what you need?

**Comment by pquerner** [comment]
192.168.0.12/30 netmask is 255.255.255.252
2001:0DB8:ABCD:0012:0000:0000:0000:0000 12 is cidr (condensed notation 2001:DB8:ABCD:12::)

So when I enter a IPv4 with CIDR notation I want to know its netmask.
There is a snippet available [here](https://stackoverflow.com/questions/5710860/php-cidr-prefix-to-netmask/5711080#5711080), but I guess using a proven library with tests is better. 

Whats this for?
For a application (Froxlor https://github.com/Froxlor/Froxlor/pull/679) I want to know the following
- is IP valid (either v4 or v6)
- (v4) convert, if available CIDR to IP/Netmask notation
- (v6) convert any valid v6 notation to "easiest" to read (see https://www.ipv6.com/general/ipv6-addressing/)

**Comment by mlocati** [comment]
> is IP valid (either v4 or v6)

This can be done with
```php
$address = IPLib\Factory::addressFromString($string);
```
If `$string` is not a string containing a valid IP address (IPv4 or IPv6), `$address` will be null.

> (v4) convert, if available CIDR to IP/Netmask notation

I agree that this could be a nice feature. Could you open a new issue about this?

> convert any valid v6 notation to "easiest" to read

This is already possible:

```php
echo IPLib\Factory::addressFromString('2001:0DB8:ABCD:0012:0000:0000:0000:0000')->toString();
```

will print `2001:db8:abcd:12::`

**Comment by pquerner** [comment]
I've instead renamed this ticket for the "converting" stuff. Is that ok?

**Comment by mlocati** [comment]
What about #29?

It adds the `asSubnet()` method to the `Pattern` range type, and the `asPattern()` method to the `Subnet` range type.

**Comment by pquerner** [comment]
I am not sure how you get netmask `255.255.255.0` from input `192.168.0.1/24` with this PR.

**Comment by mlocati** [comment]
This pull request is not related to the netmask (that will be another pull request: I do prefer to fix/improve one thing at a time)

**Comment by pquerner** [comment]
I see. Is it still needed to create a nice issue? Since you've declared the close of this issue in #29. :)

**Comment by mlocati** [comment]
Nope, don't worry. But next time please submit an issue for every single request ;)

**Comment by mlocati** [comment]
#30 implements the other request (determining the subnet mask of an IP range)

**Comment by mlocati** [comment]
I just released version 1.8.0 with the changes of #29 and #30
